src/Microsoft.DotNet.Interactive.Tests/PackageRestoreContextTests.cs

```diff
@@ -110,6 +110,25 @@ public async Task Packages_from_previous_requests_are_not_returned_in_subsequent
                         .NotContain(r => r.PackageName == "FluentAssertions");
         }
 
+        [Fact]
+        public async Task Invalid_package_restores_are_not_remembered()
+        {
+            using var restoreContext = new PackageRestoreContext();
+
+            // This package does not exist
+            restoreContext.GetOrAddPackageReference("NonExistentNugetPackage", "99.99.99-NoReallyIDontExist");
+            await restoreContext.RestoreAsync();
+
+            // Even though the previous restore failed, this one should succeed
+            restoreContext.GetOrAddPackageReference("FluentAssertions", "5.7.0");
+            var result = await restoreContext.RestoreAsync();
+
+            result.ResolvedReferences
+                       .Should()
+                       .Contain(r => r.PackageName == "FluentAssertions");
```

Comment:
This assertion isn't checking that the package isn't in the set.

Reply:
What the NonExistentNugetPackage with the version # 99.99.99-NoReallyIDontExist?

Because the test is intended to verify, that a failing package restore does not cause future resolves to fail.  I suppose we could test that the first resolve failed.

Reply:
Ah, I see. A better test name might be "a failing package restore does not cause future resolves to fail". Based on the test name, I thought this was verifying that we don't re-attempt the invalid test.

Reply:
done